app/views/provider/admin/dashboards/_developers_navigation.html.slim

```diff
@@ -53,13 +53,14 @@ nav.DashboardNavigation
     // Messages
     li.DashboardNavigation-list-item
       = dashboard_collection_link 'Message',
-                                  current_account.received_messages.not_system,
+                                  received_messages_presenter.visible_messages,
```

Comment:
Takes the collection from the presenter, which is already limited to 100 elements.